### DISCUSSION TITLE
Can I SubscribePresence without Privacy Token

### DISCUSSION THREAD

**Discussion by UmutComlekci** [tulir/whatsmeow#459] [WhatsApp protocol Q&A]
Hello,
I am trying to subscribe user presence but user is not in my contacts or I have not any conversation with him/her. I am updating my presence as `available` and subscribe to user presence but WhatsappClient write logs something like; `[Client DEBUG] Trying to subscribe to presence of <number>@s.whatsapp.net without privacy token`. 

It's looks like I subscribed to user presence but I haven't receive any `Presence` events on EventHandler. What do you think about this situation? Do I need to chat with him/her at least one time or somehow can I create/request privacy token from Whatsapp?

`PutPrivacyTokens` called from two places in the source code. Maybe we can get it from somewhere else https://github.com/tulir/whatsmeow/blob/69534bf768bc20558671acc0ba22685fe65ee60f/store/store.go#L123

**Comment by ghost** [comment]
Do you fix this problem? @UmutComlekci
